### PR TITLE
Integration with junit-realtime-plugin

### PR DESCRIPTION
**Pull request by dcendents**
Integration with junit-realtime-plugin PR jenkinsci/junit-realtime-test-reporter-plugin#33

Preview:
Screenshot from 2022-01-10 20-20-48

@timja Make sure in the build monitor view configuration to select `Display Junit Realtime progress` (last option)

Here is the pipeline for a sample project:

```groovy
node {
  parallel 'firstBranch': {
    stage('stage1') {
      realtimeJUnit('stage1_*.xml') {
        writeFile text: '''<testsuite name='a' time='4'><testcase name='a1' time='1'/><testcase name='a2' time='3'><error message='a2 failed'>a2 failed</error></testcase></testsuite>''', file: 'stage1_a.xml'
        sleep 5
        writeFile text: '''<testsuite name='b' time='6'><testcase name='b1' time='2'/><testcase name='b2' time='4' /></testsuite>''', file: 'stage1_b.xml'
        sleep 2
      }
    }
  }, 'secondBranch': {
    stage('stage2') {
      realtimeJUnit('stage2_*.xml') {
        writeFile text: '''<testsuite name='c' time='12'><testcase name='c1' time='4'/><testcase name='c2' time='5'/><testcase name='c3' time='3'/></testsuite>''', file: 'stage2_c.xml'
        sleep 8
        writeFile text: '''<testsuite name='d' time='16'><testcase name='d1' time='7'/><testcase name='d2' time='9'/></testsuite>''', file: 'stage2_d.xml'
        sleep 5
      }
    }
  }
  deleteDir()
}
```

### CODE REVIEW

File: build-monitor-plugin/src/main/resources/com/smartcodeltd/jenkinsci/plugins/buildmonitor/BuildMonitorView/configure-entries.jelly [hunk 1]
```diff
@@ -99,6 +99,10 @@
       </select>
     </f:entry>
 
+    <f:entry title="${%Display Junit Realtime progress}" help="${descriptor.getHelpFile('displayJUnitProgress')}">
```
Comment (timja) [inline build-monitor-plugin/src/main/resources/com/smartcodeltd/jenkinsci/plugins/buildmonitor/BuildMonitorView/configure-entries.jelly:102]:
```suggestion
    <f:entry title="${%Display JUnit Realtime progress}" help="${descriptor.getHelpFile('displayJUnitProgress')}">
```

**Comment by dcendents** [comment]
@basil I'd appreciate if you could review this PR, I've added unit and acceptance tests.

Otherwise I've been running this code for years so I know it works fine ;-)

**Comment by basil** [comment]
@dcendents I added a commit to remove usages of Guava. Any idea why test failures like the below [aren't failing the build](https://github.com/jan-molak/jenkins-build-monitor-plugin/pull/542/checks)?

```
Error:  features.ProjectStatusShouldBeEasyToDetermine.visualising_a_failing_project[1]  Time elapsed: 37.893 s  <<< FAILURE!
java.lang.AssertionError: 

Expected: displaying the status as 'failing'
     but: was <ProjectInformation{name=My App, status={}}>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at net.serenitybdd.screenplay.QuestionConsequence.evaluateFor(QuestionConsequence.java:49)
	at net.serenitybdd.screenplay.Actor.check(Actor.java:369)
	at net.serenitybdd.screenplay.Actor.should(Actor.java:341)
	at features.ProjectStatusShouldBeEasyToDetermine.visualising_a_failing_project(ProjectStatusShouldBeEasyToDetermine.java:65)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at net.serenitybdd.junit.runners.SerenityStatement.evaluate(SerenityStatement.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at net.serenitybdd.junit.runners.SerenityRunner.performRunChild(SerenityRunner.java:467)
	at net.serenitybdd.junit.runners.SerenityRunner.runChild(SerenityRunner.java:441)
	at net.serenitybdd.junit.runners.SerenityRunner.runChild(SerenityRunner.java:57)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at net.serenitybdd.junit.runners.SerenityRunner.run(SerenityRunner.java:250)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at net.serenitybdd.junit.runners.SerenityParameterizedRunner.run(SerenityParameterizedRunner.java:217)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:290)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:377)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:138)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:465)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:451)
```

**Comment by basil** [comment]
I see, they are flakes:

```
Warning:  features.BuildMonitorShouldBeEasyToSetUp.adding_project_to_an_empty_build_monitor[1]
Error:    Run 1: BuildMonitorShouldBeEasyToSetUp.adding_project_to_an_empty_build_monitor:52 
Expected: an element that is currently visible
     but: no matching element found by not so was not visible
Error:    Run 2: BuildMonitorShouldBeEasyToSetUp.adding_project_to_an_empty_build_monitor:52 
Expected: an element that is currently visible
     but: no matching element found by not so was not visible
[INFO]   Run 3: PASS

Warning:  features.ProjectStatusShouldBeEasyToDetermine.visualising_a_failing_project[1]
Error:    Run 1: ProjectStatusShouldBeEasyToDetermine.visualising_a_failing_project:65 
Expected: displaying the status as 'failing'
     but: was <ProjectInformation{name=My App, status={}}>
Error:    Run 2: ProjectStatusShouldBeEasyToDetermine.visualising_a_failing_project:65 
Expected: displaying the status as 'failing'
     but: was <ProjectInformation{name=My App, status={}}>
[INFO]   Run 3: PASS
```

**Comment by dcendents** [comment]
Actually the failsafe configuration will re run failing tests a second time. It is a common practice with UI testing with serenity/selenium.

**Pull request merged**